src/Microsoft.TestPlatform.CrossPlatEngine/Execution/RunTestsWithSources.cs

```diff
@@ -203,5 +206,21 @@ private static string TestCaseFilterToShow(string testCaseFilter)
 
             return testCaseFilterToShow;
         }
+
+        protected override void SendSessionEnd()
+        {
+            this.testCaseEventsHandler?.SendSessionEnd();
+        }
+
+        protected override void SendSessionStart()
+        {
+            var properties = new Dictionary<string, object>();
+
+            IEnumerable<string> sources = new List<string>();
```

Comment:
Coming at lot of places. Lets move to some common or utility class. Add comments as well of their working